pkg/controller/utils/utils.go

```diff
@@ -244,3 +245,131 @@ func GetAmazonCloudIntegration(ctx context.Context, client client.Client) (*oper
 
 	return instance, nil
 }
+
+// GetK8sEndpoint returns an ip:port of a kubernetes endpoint
+func GetK8sEndpoint(c client.Client) (string, int, error) {
+	var (
+		ip   string
+		port int
+		err  error
+	)
+
+	eps := &corev1.Endpoints{}
+	err = c.Get(context.Background(), client.ObjectKey{Name: "kubernetes", Namespace: "default"}, eps)
```

Comment:
This looks like it's using the Kubernetes API to query the k8s endpoints? Why do we need to query an endpoint if we already have API access :D

Reply:
man, I think I pushed a wrong branch! I will close this PR